### PR TITLE
Implement tests for Directory

### PR DESCRIPTION
**Pull request by tvolkert**

### CODE REVIEW

**Comment by tvolkert** [comment]
@cbracken @johnmccutchan @matanlurey 

Don't forget to "Load diff" on the actual tests :-)

File: lib/src/backends/memory/memory_file_system_entity.dart [hunk 1]
```diff
@@ -113,7 +113,10 @@ abstract class _MemoryFileSystemEntity implements FileSystemEntity {
   FileSystemEntity get absolute {
     String absolutePath = path;
     if (!_isAbsolute(absolutePath)) {
-      absolutePath = '${fileSystem._cwd}$_separator$absolutePath';
+      String base = fileSystem._cwd;
+      absolutePath = base == _separator
+          ? '$_separator$absolutePath'
```
Comment (johnmccutchan) [inline lib/src/backends/memory/memory_file_system_entity.dart]: Consider using the path package for joining paths

Comment (matanlurey) [inline lib/src/backends/memory/memory_file_system_entity.dart]: +1

Comment (tvolkert) [inline lib/src/backends/memory/memory_file_system_entity.dart]: Done (and elsewhere)

**Pull request merged**